File: src/main/java/com/temporal/agentic/workflows/NodeChildWorkflow.java
package com.temporal.agentic.workflows;

import com.temporal.agentic.models.NodeExecutionRequest;
import com.temporal.agentic.models.NodeExecutionResponse;
import io.temporal.workflow.WorkflowInterface;
import io.temporal.workflow.WorkflowMethod;

@WorkflowInterface
public interface NodeChildWorkflow {

    @WorkflowMethod
    NodeExecutionResponse executeNode(NodeExecutionRequest request);
}

File: src/main/kotlin/com/temporal/agentic/workflows/NodeChildWorkflow.kt
package com.temporal.agentic.workflows

import com.temporal.agentic.models.NodeExecutionRequest
import com.temporal.agentic.models.NodeExecutionResponse
import io.temporal.workflow.WorkflowInterface
import io.temporal.workflow.WorkflowMethod

@WorkflowInterface
interface NodeChildWorkflow {
    
    @WorkflowMethod
    fun executeNode(request: NodeExecutionRequest): NodeExecutionResponse
}
